integrations/anthropic/src/actions/generate-content.ts

```diff
@@ -0,0 +1,261 @@
+      throw new InvalidPayloadError('`toolCalls` must contain at least one tool call')
+    }
+
+    return message.toolCalls.map(
+      (toolCall) =>
+        <Anthropic.ToolUseBlockParam>{
+          type: 'tool_use',
+          id: toolCall.id,
+          name: toolCall.function.name,
+          input: toolCall.function.arguments,
+        }
+    )
+  } else if (message.type === 'tool_result') {
+    return [
+      <Anthropic.ToolResultBlockParam>{
+        type: 'tool_result',
+        tool_use_id: message.toolResultCallId,
+        content: [
+          {
+            type: 'text',
+            text: message.content as string,
+          },
+        ],
+      },
+    ]
+  } else {
+    throw new InvalidPayloadError(`Message type "${message.type}" is not supported`)
+  }
+}
+
+function mapToAnthropicTools(input: llm.schemas.GenerateContentInput): Anthropic.Tool[] | undefined {
+  if (input.toolChoice?.type === 'none') {
+    // Don't return any tools if tool choice was to not use any tools
+    return []
+  }
+
+  const anthropicTools = input.tools?.map(
+    (tool) =>
+      <Anthropic.Tool>{
+        name: tool.function.name,
+        description: tool.function.description,
+        input_schema: tool.function.inputSchema,
+      }
+  )
+
+  // note: don't send an empty tools array
+  return anthropicTools?.length ? anthropicTools : undefined
+}
+
+function mapToAnthropicToolChoice(
+  toolChoice: llm.schemas.GenerateContentInput['toolChoice']
+): Anthropic.MessageCreateParams['tool_choice'] {
+  if (!toolChoice) {
+    return undefined
+  }
+
+  switch (toolChoice.type) {
+    case 'any':
+      return <MessageCreateParams.ToolChoiceAny>{ type: 'any' }
+    case 'auto':
+      return <MessageCreateParams.ToolChoiceAuto>{ type: 'auto' }
+    case 'specific':
+      return <MessageCreateParams.ToolChoiceTool>{
+        type: 'tool',
+        name: toolChoice.functionName,
+      }
+    case 'none': // This is handled when passing the tool list by removing all tools, as Anthropic doesn't support a "none" tool choice
+    default:
+      return undefined
+  }
+}
+function mapToStopReason(
+  anthropicStopReason: Anthropic.Message['stop_reason']
+): llm.schemas.GenerateContentOutput['choices'][0]['stopReason'] {
+  switch (anthropicStopReason) {
+    case 'end_turn':
+    case 'stop_sequence':
+      return 'stop'
+    case 'max_tokens':
+      return 'max_tokens'
+    case 'tool_use':
+      return 'tool_calls'
+    default:
+      return 'other'
+  }
+}
+
+function mapToToolCalls(response: Anthropic.Message): llm.schemas.ToolCall[] | undefined {
+  const toolCalls = response.content.filter((x): x is Anthropic.ToolUseBlock => x.type === 'tool_use')
+
+  return toolCalls.map((toolCall) => {
+    return {
+      id: toolCall.id,
+      type: 'function',
+      function: {
+        name: toolCall.name,
+        arguments: toolCall.input as llm.schemas.ToolCall['function']['arguments'],
+      },
+    }
+  })
+}
```

Comment:
:bulb: **suggestion (non-blocking):** maybe we could add tests for LLMs content conversion? otherwise updates might be a bit trickier

Reply:
Totally agree. My plan was to get this out of the door asap, and I'll add a Linear issue so I can add tests for this integration as soon as possible.